csrc/mamba/causal_conv1d/causal_conv1d.cu

```diff
@@ -309,19 +320,38 @@ void causal_conv1d_fwd_kernel(ConvParamsBase params) {
     auto& smem_store_vec = reinterpret_cast<typename Ktraits::BlockStoreVecT::TempStorage&>(smem_);
     vec_t *smem_exchange = reinterpret_cast<vec_t *>(smem_ + Ktraits::kSmemIOSize);
 
+    const bool kVarlen = params.cu_seq_len_ptr != nullptr;
     const int tidx = threadIdx.x;
     const int batch_id = blockIdx.x;
     const int channel_id = blockIdx.y;
-    input_t *x = reinterpret_cast<input_t *>(params.x_ptr) + batch_id * params.x_batch_stride
+    const int *cu_seq_len = kVarlen ? reinterpret_cast<int *>(params.cu_seq_len_ptr) : nullptr;
+    const int sequence_start_index = kVarlen ? (batch_id == 0 ? 0 : cu_seq_len[batch_id - 1]) : batch_id;
+    const int seqlen = kVarlen ? cu_seq_len[batch_id] - sequence_start_index : params.seqlen;
```

Comment:
Do we run into any load-imbalance issues from parallelizing over the batch dim, when the sequence length is variable?

Reply:
In some sense, we do, some blocks launches will finish faster than others, but then they will be free to run other scheduled blocks. 
A better implementation than this one IMO would be that each block launch would be responsible for computing a fixed sized chunk of data. In a sense it would be more efficient one on side, but on the other side, each chunk/block would be responsible for multiple sequences which would or could complicate the implementation and would even affect the overall performances.
I implemented it in this way since its easier to understand and maintain (and easier to implement). 
but we definitely can, in the future, think about different schemes/way that will make it even better.

Reply:
OK, that sounds good, thanks for the explanation